### PR TITLE
Improving port highlighting

### PR DESCRIPTION
**Pull request by wdanilo**
### Checklist
Please include the following checklist in your PR:

- [ ] The documentation has been updated if necessary.
- [ ] All code conforms to the [Rust](https://github.com/luna/enso/blob/main/docs/style-guide/rust.md) style guide.
- [ ] All code has automatic tests where possible.
- [ ] All code has been profiled where possible.
- [ ] All code has been manually tested in the IDE.
- [ ] All code has been manually tested in the "debug/interface" scene.
- [ ] All code has been manually tested by the PR owner against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).
- [ ] All code has been manually tested by at least one reviewer against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).

### CODE REVIEW

File: src/rust/ide/view/graph-editor/src/component/node.rs [hunk 1]
```diff
@@ -148,8 +148,8 @@ ensogl::define_endpoints! {
         select              (),
         deselect            (),
         set_visualization   (Option<visualization::Definition>),
-        set_dimmed          (bool),
-        set_input_connected (span_tree::Crumbs,bool),
+        set_disabled         (bool),
```
Comment (farmaazon) [inline src/rust/ide/view/graph-editor/src/component/node.rs:151]:
```suggestion
        set_disabled        (bool),
```

File: src/rust/ide/view/graph-editor/src/component/node/input/area.rs [hunk 2]
```diff
@@ -303,6 +321,9 @@ impl Area {
 
         frp::extend! { network
 
+            trace frp.set_expression_usage_type;
+            trace frp.set_connected;
+
```
Comment (farmaazon) [inline src/rust/ide/view/graph-editor/src/component/node/input/area.rs:326]: trace left

File: src/rust/ide/view/graph-editor/src/component/node/input/area.rs [hunk 3]
```diff
@@ -328,14 +349,14 @@ impl Area {
...
+                (&frp.input.set_edit_mode,&frp.input.set_edit_ready_mode,&frp.input.set_ports_active,
+                |edit_mode,edit_ready_mode,(set_ports_active,_)|
+                     (*edit_mode || *edit_ready_mode) && !set_ports_active
                 );
-            port_vis <- all_with(&frp.input.ports_active,&edit_mode,|a,b|*a&&(!b));
+            port_vis <- all_with(&frp.input.set_ports_active,&edit_mode,|(a,_),b|*a&&(!b));
 
             frp.output.source.ports_visible <+ port_vis;
-            frp.output.source.editing       <+ edit_mode.sampler();
+            frp.output.source.editing       <+ edit_mode.sampler(); // FIXME do we need sampler here?
```
Comment (farmaazon) [inline src/rust/ide/view/graph-editor/src/component/node/input/area.rs:359]: FIXME left

File: src/rust/ide/view/graph-editor/src/component/node/input/area.rs [hunk 4]
```diff
@@ -480,6 +500,7 @@ impl Area {
                 || builder.parent_parensed;
 
             if let Some(id) = node.ast_id {
+                println!("New id mapping: {} -> {:?}",id,node.crumbs);
```
Comment (farmaazon) [inline src/rust/ide/view/graph-editor/src/component/node/input/area.rs:503]: debug print left

File: src/rust/ide/view/graph-editor/src/component/node/input/area.rs [hunk 5]
```diff
@@ -683,21 +704,26 @@ impl Area {
...
             def_tp.emit(node.tp().cloned().map(|t|t.into()));
-            Some(frp.final_type.clone_ref().into())
+            Some(frp.tp.clone_ref().into())
         });
     }
 
     pub(crate) fn set_expression(&self, expression:impl Into<node::Expression>) {
         let model          = &self.model;
         let expression     = expression.into();
+        println!("set_expression {:?}",expression);
```
Comment (farmaazon) [inline src/rust/ide/view/graph-editor/src/component/node/input/area.rs:726]: debug print left.

File: src/rust/ide/view/graph-editor/src/lib.rs [hunk 6]
```diff
@@ -1192,20 +1204,25 @@ impl GraphEditorModel {
             }
 
             if let Some(target) = edge.take_target() {
-                self.set_input_connected(&target,false);
+                self.set_input_connected(&target,None,false); // FIXME None
```
Comment (farmaazon) [inline src/rust/ide/view/graph-editor/src/lib.rs:1207]: Fixme left.

File: src/rust/ide/view/graph-editor/src/lib.rs [hunk 7]
```diff
@@ -1991,21 +2046,18 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
     edge_target_click <- valid_edge_disconnect_click.gate_not(&edge_is_source_click);
 
     // FIXME: make nicer - out.source.edge_source_unset should contain information from which node it ws unset
-    eval edge_target_click((edge_id)
-        model.with_edge_target(edge_id.0,|t| model.set_input_connected(&t,false)));
+    eval edge_target_click((edge_id) model.set_edge_target_connected(edge_id.0,false));
+        // model.with_edge_target(edge_id.0,|t| model.set_input_connected(&t,false)));
```
Comment (farmaazon) [inline src/rust/ide/view/graph-editor/src/lib.rs:2050]: Commented code left.

File: src/rust/ide/view/graph-editor/src/lib.rs [hunk 8]
```diff
@@ -2121,6 +2175,7 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
     out.source.edge_added      <+ edge;
     out.source.edge_source_set <+ new_edge_source;
     out.source.edge_target_set <+ new_edge_target;
+    trace out.edge_target_set;
```
Comment (farmaazon) [inline src/rust/ide/view/graph-editor/src/lib.rs:2178]: trace left.

File: src/rust/ide/view/graph-editor/src/lib.rs [hunk 9]
```diff
@@ -2484,6 +2539,9 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
 
     eval out.edge_source_set        (((id,tgt)) model.set_edge_source(*id,tgt));
     eval out.edge_target_set        (((id,tgt)) model.set_edge_target(*id,tgt));
+    // todo
+    eval out.edge_target_set (((edge_id,_)) model.set_edge_target_connected(*edge_id,true));
+
```
Comment (farmaazon) [inline src/rust/ide/view/graph-editor/src/lib.rs:2544]: uknown todo left.

**Review by farmaazon — APPROVED** [review]

File: src/rust/ide/view/graph-editor/src/lib.rs (hunk 6 above)
Comment (wdanilo) [reply src/rust/ide/view/graph-editor/src/lib.rs:1207]: Will be resolved in next pr

**Pull request merged**